ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -272,12 +273,13 @@ void setFlag(Flags flag) {
    * @param newOffset the new {@link Offset} to be updated for the {@link IndexValue}
    */
   void setNewOffset(Offset newOffset) {
```

Comment:
I want to keep the code in `BlobStoreCompactor` similar to what is available in `markAsDeleted()` - if we want to change that,  we can do so as a part of another change.